app/controllers/teams_controller.rb

```diff
@@ -35,9 +35,13 @@ def update
   end
 
   def destroy
-    return unless @team.destroy
-    flash[:success] = 'Success'
-    redirect_to teams_path
+    if @team.service_providers.empty? && @team.destroy
+      flash[:success] = 'Success'
+      redirect_to teams_path and return
```

Comment:
This may be on the personal preference side so feel free to take or leave. I find `return redirect_to(team_path)` a little more readable.